### PR TITLE
SNS - Create `MockI2c` (and temperature sensor tests)

### PR DESCRIPTION
**Pull request by davidbeechey**
Creates `MockI2c` for use in testing. Temperature sensor (STTS22H) basic tests also created to try it out (closes HYPE-53).

This changes the way `HypedI2c` is passed into `Temperature::new()`, it is now a mutable reference to the `HypedI2c` instance.

### CODE REVIEW

File: lib/io/src/i2c.rs [hunk 1]
```diff
@@ -22,3 +21,40 @@ pub trait HypedI2c {
...
+
+    impl crate::i2c::HypedI2c for MockI2c {
+        /// Reads a byte by looking up the device address and register address in the map
+        fn read_byte(&mut self, device_address: u8, register_address: u8) -> Option<u8> {
+            self.values
+                .get(&(device_address, register_address))
+                .copied()
+        }
+
+        /// Always succeeds, but does nothing
```
Comment (TomLonergan03) [inline lib/io/src/i2c.rs]: something you could do here if you care is store the writes that occur and have a getter, then you could test that the correct writes occur

File: lib/io/src/i2c.rs [hunk 2]
```diff
@@ -22,3 +21,40 @@ pub trait HypedI2c {
         data: u8,
     ) -> Result<(), I2cError>;
 }
+
+pub mod mock_i2c {
+    use heapless::FnvIndexMap;
+
+    /// A fixed-size map of I2C values, indexed by device address and register address
+    type I2cValues = FnvIndexMap<(u8, u8), u8, 16>;
```
Comment (TomLonergan03) [inline lib/io/src/i2c.rs]: can the return type be an `Option<u8>`? then you can test failure cases as well

File: lib/io/src/i2c.rs (hunk 1 above)
Comment (TomLonergan03) [reply lib/io/src/i2c.rs]: and/or have a list of write results, so you can check failure cases again

**Review by TomLonergan03 — APPROVED** [review]
looks good, a couple suggestions for how you could make the mock more coveringer

**Review by H-Allen — APPROVED** [review]

**Pull request merged**